src/Authentication.Abstractions/Interfaces/IClientFactory.cs

```diff
@@ -38,6 +39,17 @@ public interface IClientFactory: IHyakClientFactory
         /// targeting the given named endpoint in the targeted environment</returns>
         TClient CreateArmClient<TClient>(IAzureContext context, string endpoint) where TClient : ServiceClient<TClient>;
 
+        /// <summary>
+        /// Create a properly configured AutoRest client using the given target Azure context, named endpoint and cmdlet context called from
+        /// </summary>
+        /// <typeparam name="TClient">The client type to create</typeparam>
+        /// <param name="context">The azure context to target</param>
+        /// <param name="endpoint">The named endpoint the client shoulld target</param>
+        /// <param name="cmdletContext">The cmdlet context called from</param>
+        /// <returns>A client properly authenticated in the given context, properly configured for use with Azure PowerShell, 
+        /// targeting the given named endpoint in the targeted environment</returns>
+        TClient CreateArmClient<TClient>(IAzureContext context, string endpoint, ICmdletContext cmdletContext) where TClient : ServiceClient<TClient>;
```

Comment:
Could you explain why creating an ArmClient needs cmdletContext?

Reply:
A lot of partners use Arm Client to handle the authentication. If we want to collect authentication telemetry from their cmdlets, a good way is to add cmdlet context here.
Also the CmdletContext can be treated as the same status as AzureContext. We can pass them both in the same interface.